controllers/auth.controller.ts

```diff
@@ -0,0 +1,83 @@
+import express, { NextFunction, Request, Response, Router } from "express";
+import { ErrorMessage } from '../models/errors';
+import { LoginRequestBody, RegisterRequestBody } from '../models/request.body';
+import {
+    LoginResponseBody,
+    LoginResponseData,
+    RegisterResponseBody,
+    RegisterResponseData
+} from '../models/response.body';
+import { UserEntity } from '../models/user.entity';
+import { getUserByEmail } from '../repositories/user.repository';
+import { getUserByEmailAndPassword, loginUser, registerUser } from '../services/user.service';
+
+export const authRouter: Router = express.Router();
+
+authRouter.post('/register', async (req: Request<any>, res: Response<RegisterResponseBody>, next: NextFunction) => {
+    try {
+        const user: RegisterRequestBody = req.body;
+
+        const userByEmail: UserEntity | null = await getUserByEmail(user.email);
+
+        if (userByEmail) {
+            return res.status(400)
+                .send({
+                    "data": null,
+                    "error": {
+                        "message": ErrorMessage.NotValidEmail
```

Comment:
The given code snippet is not very clear. Either rename the variable to something like **emailExists** or provide an error indicating that the email is already registered.